### PR TITLE
Upsert response

### PR DESCRIPTION
**Pull request by mrgaaron**
Handling for the upsert response given by write datapoints

### CODE REVIEW

File: src/main/java/com/tempoiq/UpsertResponse.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    HashMap<String, DeviceStatus> results = new HashMap<String, DeviceStatus>();
+    for (Map.Entry<String, DeviceStatus> entry : this.statuses.entrySet()) { 
+      if (entry.getValue().getSuccess() == false) {
+        results.put(entry.getKey(), entry.getValue());
+      }
+    }
+    return results;
+  }
+  
+  public static UpsertResponse make(HttpResponse response) throws java.io.IOException {
```
Comment (jcc333) [inline src/main/java/com/tempoiq/UpsertResponse.java]: Why isn't this just a constructor overload?

Comment (mrgaaron) [inline src/main/java/com/tempoiq/UpsertResponse.java]: It's used by Result to construct toys dynamically through reflection so I
had to match the method name and signature it expects

Comment (mrgaaron) [inline src/main/java/com/tempoiq/UpsertResponse.java]: Objects not toys.  Stupid swypos

File: src/main/java/com/tempoiq/UpsertResponse.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  public boolean wasPartiallySuccessful() {
+    int failures = 0;
+    Set<Map.Entry<String, DeviceStatus>> entries = this.statuses.entrySet();
+    for (Map.Entry<String, DeviceStatus> entry : entries) { 
+      if (entry.getValue().getSuccess() == false) {
+        failures += 1;
+      }
+    }
+    if (failures < entries.size()) {
```
Comment (jcc333) [inline src/main/java/com/tempoiq/UpsertResponse.java]: this if/else is redundant; you can just `return failures < entries.size();`

File: src/main/java/com/tempoiq/UpsertResponse.java (hunk 1 above)
Comment (jcc333) [inline src/main/java/com/tempoiq/UpsertResponse.java]: cool cool

**Pull request merged**